pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -206,8 +207,19 @@ func getPhase2Binding(vmi *v1.VirtualMachineInstance, iface *v1.Interface, netwo
 		return nil
 	}
 
+	var tapName = ""
+	// when running on virt-handler, we got to create the tap device
+	if domain == nil && (iface.Bridge != nil || iface.Masquerade != nil) {
+		tapDeviceName, err := createTapDevice()
```

Comment:
IMO, the creation of the tap device should be in `preparePodNetworkInterfaces` method (same as the bridge creation). I would expect `getPhase2Binding' to only return the binding mechanism and not creating devices in virt-launcher.

Reply:
Now I see it is moved to `preparePodNetworkInterfaces` in the next commit. So ignore:)

Reply:
Sorry for the overall structure of the PR, and enduring it.

I'll re-work it, squashing some stuff together, which will hopefully make it a lot easier for the next reviewer.